extra_data/filecache.py

```diff
@@ -0,0 +1,125 @@
+"""
+File cache implementation
+
+File cache provides the collection for open h5 files 
+which keeps the number of opened files under the given limit
+popping the files that were accessed the longest time ago
+
+Implementation based on the OrderedDict
+
+The module provides global instance of file cache for using
+across entire process
+"""
+import h5py
+from collections import OrderedDict
+
+class FileRef(object):
+    __slots__ = 'nref', 'fh'
+
+
+class FileCache(object):
+    """
+    FileCache is a collection for opened h5 files. It keeps the number of opened files 
+    under the given limit popping files accessed longest time ago.
+    """
+    def __init__(self, maxfiles=128):
+        self._maxfiles = maxfiles
+        self._cache = OrderedDict()
+        
+    def __del__(self):
+        self.clean()
```

Comment:
Is a `__del__` method necessary? In normal usage there will only be one `FileCache` and it will last until the process terminates, right?

Reply:
I prefer to keep. This doesn't ask for food but will work in a strange but possible situation if someone decided to delete and recreate the `Filecache`.